packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -1244,9 +1270,9 @@ export function warnForInsertedHydratedElement(
     }
     didWarnInvalidHydration = true;
     console.error(
-      'Expected server HTML to contain a matching <%s> in <%s>.',
+      'Expected server HTML to contain a matching <%s> in %s.',
```

Comment:
Is it clear what the "in" really means here? That's the thing that I always found vague. Like are these the mismatches, or is one parent and one child?

I feel like the message you use some formatting.

Reply:
Yeah, I'd propose changing it more significantly. Something like

```
There is a mismatch between the client and server render output.
The mismatch occurred inside of this parent element:

  <div class="blabla parent">

The last successfully matched child was:

  <h1 class="blabla">

After it, the server rendered:

  <h3 class="blabla">

But the client rendered:

  <h2 class="blabla">

Fix the error by making the client and the server output the same.
```

Thoughts?

Reply:
```
There is a mismatch between the client and server render output.
The mismatch occurred inside of this element:

  <div class="blabla parent">
    <h1 class="blabla">
-   <h3 class="blabla">  <--- server
+   <h2 class="blabla">  <--- client
```

Reply:
The same format translates to missing and extra nodes too.